text/0499-appconfig-constructs.md

```diff
@@ -0,0 +1,422 @@
+    ActionPoint.ON_DEPLOYMENT_COMPLETE,
+  ],
+});
+```
+
+## Higher level resources
+
+This section details how we will implement our constructs that combine multiple different resources into one.
+
+# Configuration
+
+A configuration is a higher level construct that can either be a HostedConfiguration (stored internally through AppConfig) or a SourcedConfiguration (stored in S3, Secrets Manager, SSM Parameter, SSM Document, or Code Pipeline). This construct will be used as a building block to deploy configuration through the AppConfig construct (detailed in the next section). For a HostedConfiguration, this construct will combine both a configuration profile and a hosted configuration version.
+
+### Example
+```ts
+const hostedConfig = new HostedConfiguration(this, 'MyHostedConfig', {
+  content: 'This is the configuration content.',
+  application: <IApplication>,      // only required if creating new configuration profile
+
+  // optional, taken directly from ConfigurationProfile props
+  name: '...',
+  tags: '...',
+  validators: '...',
+  configurationType: '...',
+  configurationProfile: <IConfigurationProfile>,    // if specified, we will use this config profile instead of creating one
+
+  // optional, taken directly from HostedConfigurationVersion props
+  versionLabel: '...',
+  latestVersionNumber: 1,
+  contentType: '...',
+
+  // optional
+  description: '...',   // description for both resources
+  deployTo: [           // will deploy to all environments by default
+    Environment.create('beta'),
+    Environment.create('prod'),
+    Environment.from(<IEnvironment>),
+  ],
+});
+
+const sourcedConfig = new SourcedConfiguration(this, 'MySourcedConfig', {
+  location: <IBucket|IParameter|IDocument|ISecret|IPipeline>,   // ex. Location.fromBucket(<IBucket>), only required if creating new configuration profile
+  application: <IApplication>,                                  // only required if creating new configuration profile
+  versionNumber: '...',
+
+  // optional, taken directly from ConfigurationProfile props
+  name: '...',
+  retrievalRole: '...',
+  validators: '...',
+  configurationType: '...',
+  description: '...',
+  tags: '...',
+  configurationProfile: <IConfigurationProfile>,    // if specified, we will use this config profile instead of creating one
+
+  // optional
+  deployTo: [    // will deploy to all environments by default
+    Environment.create('beta'),
+    Environment.create('prod'),
+    Environment.from(<IEnvironment>),
+  ],
+});
+```
+
+We will also be able to add these configurations to an application by calling `addHostedConfiguration` or `addSourcedConfiguration`.
+
+# AppConfig
+
+An AppConfig construct will be the simplest way to create and deploy configuration. This construct will handle deployments for you and start deploying configuration on creation.
+
+### Example
+```ts
+const appconfig = new AppConfig(this, 'MyAppConfig',{
+  deploymentStrategy: <IDeploymentStrategy>,
```

Comment:
Other than `AppConfig`, the `DeploymentStrategy` will not be associated to anything else, however it could be potentially associated with a `CfnDeployment` from our L1 constructs. The `DeploymentStrategy` construct is only relevant when we are deploying configuration which the `AppConfig` construct (and `CfnDeployment` if users want) will handle. Does this answer your question of how is it consumed?

Reply:
It does. By the way, how does `AWS::AppConfig::Deployment` work? Do you have to modify its logical ID every time you make a change to make sure your change is actually published (a la API Gateway)?

Reply:
You don't have to modify the logical ID, it looks like when I redeploy a `AWS::AppConfig::Deployment`, it will create a new resource in CFN but it will still retain the old deployment